File: internal/indexer/testdata/go/services/critical_user_model_service.go
package services

import "models"

// StoreCriticalUserModel persists a CriticalUserModel.
func StoreCriticalUserModel(m models.CriticalUserModel) bool {
	return m.Name != ""
}

// LoadCriticalUserModel retrieves a CriticalUserModel by ID.
func LoadCriticalUserModel(id int) models.CriticalUserModel {
	return models.CriticalUserModel{Name: "loaded", Email: "loaded@example.com"}
}

// CriticalUserModelRepository manages CriticalUserModel persistence.
type CriticalUserModelRepository struct {
	models []models.CriticalUserModel
}

// Add stores a CriticalUserModel in the repository.
func (r *CriticalUserModelRepository) Add(m models.CriticalUserModel) {
	r.models = append(r.models, m)
}

File: internal/indexer/testdata/rust/services/critical_user_model_service.rs
use crate::models::critical_user_model::CriticalUserModel;

/// CriticalUserModelRepository manages CriticalUserModel persistence.
pub struct CriticalUserModelRepository {
    models: Vec<CriticalUserModel>,
}

impl CriticalUserModelRepository {
    pub fn new() -> CriticalUserModelRepository {
        CriticalUserModelRepository { models: Vec::new() }
    }

    /// Saves a CriticalUserModel to the repository.
    pub fn save(&mut self, model: CriticalUserModel) {
        self.models.push(model);
    }

    /// Finds a CriticalUserModel by email.
    pub fn find_by_email(&self, email: &str) -> Option<&CriticalUserModel> {
        self.models.iter().find(|m| m.email == email)
    }
}

/// store_critical_user_model is a standalone function for persistence.
pub fn store_critical_user_model(model: CriticalUserModel) -> bool {
    !model.name.is_empty()
}
